contracts/script/deploy/eth_main/Alpha_init_deploy.s.sol

```diff
@@ -0,0 +1,501 @@
+        );
+
+        //deploy serviceManager
+        {
+            uint64 recurrentRegistrationLimit =
+                uint64(stdJson.readUint(configData, ".registryParams.recurrentRegistrationLimit"));
+
+            serviceManagerImplementation = new FinalizerServiceManager(
+                avsDirectory,
+                rewardsCoordinator,
+                registryCoordinator,
+                stakeRegistry,
+                IFinalizerTaskManager(address(taskManager)),
+                recurrentRegistrationLimit
+            );
+        }
+
+        // upgrade service manager proxy to implementation and initialize
+        avsProxyAdmin.upgradeAndCall(
+            TransparentUpgradeableProxy(payable(address(serviceManager))),
+            address(serviceManagerImplementation),
+            abi.encodeWithSelector(serviceManager.initialize.selector, avsOwner, avsOwner, ejector)
+        );
+
+        // deploy RegistryCoordinator
+        registryCoordinatorImplementation = new RegistryCoordinator(
+            IServiceManager(address(serviceManager)), stakeRegistry, blsApkRegistry, indexRegistry
+        );
+
+        (
+            uint96[] memory minimumStakeForQuorum,
+            IStakeRegistry.StrategyParams[][] memory strategyAndWeightingMultipliers
+        ) = _parseStakeRegistryParams(configData);
+
+        // parse initalization params and permissions from config data
+        bytes memory operatorConfigsRaw = stdJson.parseRaw(configData, ".operatorSetParams");
+        IRegistryCoordinator.OperatorSetParam[] memory operatorSetParams =
+            abi.decode(operatorConfigsRaw, (IRegistryCoordinator.OperatorSetParam[]));
+
+        // upgrade registry coordinator proxy to implementation and initialize
+        avsProxyAdmin.upgradeAndCall(
+            TransparentUpgradeableProxy(payable(address(registryCoordinator))),
+            address(registryCoordinatorImplementation),
+            abi.encodeWithSelector(
+                RegistryCoordinator.initialize.selector,
+                avsOwner,
+                churner,
+                address(serviceManager),
+                avsPauserReg,
+                0,
+                operatorSetParams,
+                minimumStakeForQuorum,
+                strategyAndWeightingMultipliers
+            )
+        );
+
+        blsSignatureChecker = new BLSSignatureChecker(registryCoordinator);
+
+        operatorStateRetreiverExtended = new OperatorStateRetrieverExtended();
+
+        taskManagerImplementation = new FinalizerTaskManager();
+        taskManagerImplementation.initialize(
+                avsPauserReg,
+                avsOwner,
+                aggregator,
+                aggregator,
+                allowNonRootTmInit,
+                address(blsSignatureChecker),
+                taskResponseWindowBlocks,
+                address(operatorStateRetreiverExtended),
+                rolldown
+        );
```

Comment:
we don't have a proper constructor with `disableInitializers` 
calling this here to disable init calls on the implementation contract since it can be called only once
also ensures that verify contract pointers pass

same for rolldown down below